src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/DeprecationPackageMetadataCapability.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Protocol.Model;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI.Models.Package
+{
+    internal class DeprecationPackageMetadataCapability : IDeprecationCapable
+    {
+        private readonly IPackageMetadataRetrievalAdapter _packageMetadataRetrievalAdapter;
+        private PackageDeprecationMetadataContextInfo? _deprecationMetadata;
+
+        public DeprecationPackageMetadataCapability(IPackageMetadataRetrievalAdapter packageMetadataRetrievalAdapter)
+        {
+            _packageMetadataRetrievalAdapter = packageMetadataRetrievalAdapter ?? throw new ArgumentNullException(nameof(packageMetadataRetrievalAdapter));
+        }
+
+        public AlternatePackageMetadataContextInfo? AlternatePackage => _deprecationMetadata?.AlternatePackage;
+
+        public bool IsDeprecated => _deprecationMetadata != null;
+
+        public PackageDeprecationReasonEnum PackageDeprecationReasons
+        {
+            get
+            {
+                if (_deprecationMetadata?.Reasons == null || !_deprecationMetadata.Reasons.Any())
```

Comment:
```suggestion
                if (_deprecationMetadata?.Reasons == null || _deprecationMetadata.Reasons.Count == 0)
```

We should be in the habit of avoiding LINQ's `.Any()` wherever possible.

Reply:
Done, thanks!